src/Sentry/Internal/Extensions/StreamExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Sentry.Internal.Extensions
+{
+    internal static class StreamExtensions
+    {
+        public static async Task<int> ReadByteAsync(
+            this Stream stream,
+            CancellationToken cancellationToken = default)
+        {
+            using var buffer = new PooledBuffer<byte>(1);
```

Comment:
What about stackalloc instead? Assign to Span and won't need unsafe.

Reply:
Would it be beneficial? With pooling we will completely avoid allocations.

Reply:
Another concern is that it's an async method, so it might not be possible to use. I will take a look and try.

Reply:
![image](https://user-images.githubusercontent.com/1935960/97367563-0611b080-18b2-11eb-9fe8-79a265287368.png)

Looks like it can't. And `stackalloc` can't be assigned to `Memory<byte>`.

Reply:
Fuck